components/molecule/drawer/src/Overlay.js

```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types'
+import {useEffect, forwardRef} from 'react'
+
+const DrawerOverlay = forwardRef(
+  ({onClick, isVisible, target, children, ...props}, forwardedRef) => {
+    useEffect(() => {
+      if (target !== undefined) {
+        target.current.style.position = 'relative'
+        target.current.style.overflow = 'hidden'
+      }
+    }, [target])
+    return isVisible ? (
+      <div
+        ref={forwardedRef}
+        className="react-MoleculeDrawer-overlay"
+        onClick={event => {
```

Comment:
Why do you need this? can onClick be something different than a function? otherwise you don't need to do this check and you can directly spread it with the other props.